### DISCUSSION TITLE
What is the problem when multi-signing the wallet?

### DISCUSSION THREAD

**Discussion by apengn** [filecoin-project/lotus#5566] [Storage Provider] [answered]
**Describe the bug**
What is the problem when multi-signing the wallet

**To Reproduce**
1. Run './lotus msig propose --from=t3sjg2wi3fgwnc7axac7z2sol5nogzxi647hbalpauw4lf7bmm6jfgbfmhx625bcyz2trypzetwlhdgiblhpca f01001 t3qw23tzti65d7q2g4k2jtsgravrqkaltzxq36mebbd6g7zfohn2awpuiognbr3hhukseda5bqxotvttd7zihq 2;
lotus msig approve -from=t3sjg2wi3fgwnc7axac7z2sol5nogzxi647hbalpauw4lf7bmm6jfgbfmhx625bcyz2trypzetwlhdgiblhpca f01001 1 t3rmobekzwc5wd5vr6oyephxobbvrrdvxexyhpb64yhycpv2hkwxmlrwhfrq7phbhhbl7whbjkbiav54cta4ta t3qw23tzti65d7q2g4k2jtsgravrqkaltzxq36mebbd6g7zfohn2awpuiognbr3hhukseda5bqxotvttd7zihq  2'
2. See error
```
ERROR: GasEstimateMessageGas error: estimating gas used: message execution failed: exit 16, reason: hash does not match proposal params (ensure requester is an ID address) (RetCode=16)
```
**Expected behavior**
approve success

**Screenshots**

**Version (run `lotus version`):**
v1.4.1
**Additional context**
![image](https://user-images.githubusercontent.com/8267975/107388133-77f3ca00-6b30-11eb-8035-c7df8edbd829.png)

![image](https://user-images.githubusercontent.com/8267975/107387788-221f2200-6b30-11eb-8eff-19dfdca5438a.png)

**Comment by jennijuju** [comment, marked as answer]
Make sure that the the message you are approving has the same params with the originally proposed message. https://docs.filecoin.io/get-started/lotus/multisig/#propose-and-approve-a-transaction

`lotus msig approve walletAddress transactionID proposerAddress destinationAddress value`

which seems like in your `approve` message the `proposerAddress` wasn't the proposer address `t3sjg2wi3fgwnc7axac7z2sol5nogzxi647hbalpauw4lf7bmm6jfgbfmhx625bcyz2trypzetwlhdgiblhpca`. you should also use a second signer address to approve the message

**Comment by jc0803kevin** [comment]
从这几个方面查一下问题
* txid 多签交易id不对
* 对应多签交易id 的金额不对
* requester id 不对 ： 正确地址 为 propose 地址的对应id address
